searchlib/src/vespa/searchlib/attribute/address_space_usage.cpp

```diff
@@ -1,33 +1,52 @@
 // Copyright 2017 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
 
+#include "address_space_components.h"
 #include "address_space_usage.h"
-#include "i_enum_store.h"
 
 namespace search {
 
 using vespalib::AddressSpace;
 
 AddressSpaceUsage::AddressSpaceUsage()
-        : _enumStoreUsage(defaultEnumStoreUsage()),
-          _multiValueUsage(defaultMultiValueUsage()) {
+    : _map()
+{
+}
+
+AddressSpaceUsage::AddressSpaceUsage(const AddressSpace& enum_store_usage,
+                                     const AddressSpace& multi_value_usage)
+    : _map()
+{
+    // TODO: Remove this constructor and instead add usage for each relevant component explicit.
+    add(AddressSpaceComponents::enum_store, enum_store_usage);
+    add(AddressSpaceComponents::multi_value, multi_value_usage);
+}
+
+void
+AddressSpaceUsage::add(const vespalib::string& component, const vespalib::AddressSpace& usage)
+{
+    _map[component] = usage;
```

Comment:
Is there a use case where `usage` should be _added_ to any existing mapping of `component` instead of overwriting it?

Reply:
No. I will rename the function to _set()_ to better reflect the intent.

Reply:
Fixed